src/utils/subscription-registry.js

```diff
@@ -220,32 +205,44 @@ class SubscriptionRegistry {
     delayedBroadcasts.sharedMessages += msgString
     const stop = delayedBroadcasts.sharedMessages.length
 
-    // uniqueSendersMap maps from uuid to offset in uniqueSendersVector
+    const uniqueSenders = delayedBroadcasts.uniqueSenders
+    const sharedMessages = delayedBroadcasts.sharedMessages
+
     // each uniqueSender has a vector of "gaps" in relation to sharedMessage
     // senders should not receive what they sent themselves, so a gap is inserted
     // for every send from this sender
-    let pos
     if (sender) {
-      pos = delayedBroadcasts.uniqueSendersMap[sender.uuid]
-      if (pos !== undefined) {
-        delayedBroadcasts.uniqueSendersVector[pos].gaps.push({ start, stop })
-      } else {
-        pos = delayedBroadcasts.uniqueSendersVector.length
-        delayedBroadcasts.uniqueSendersMap[sender.uuid] = pos
-        delayedBroadcasts.uniqueSendersVector[pos] = {
+      const index = utils.sortedIndexBy(uniqueSenders, sender, 'uuid')
```

Comment:
NOTE: Doing a binary search on an array can be as fast or (significantally) faster than a hash map lookup (Object, Set) for smallish arrays (~10000 elements) that fit in CPU cache.